vms/platformvm/state/stakers_ops_test.go

```diff
@@ -66,73 +67,50 @@ func simpleStakerStateProperties(storeCreatorF func() (Stakers, error)) *gopter.
 			}
 
 			// no staker before insertion
-			_, err = store.GetCurrentValidator(s.SubnetID, s.NodeID) // check version 1
+			_, err = store.GetCurrentValidator(s.SubnetID, s.NodeID)
 			if err != database.ErrNotFound {
 				return fmt.Sprintf("unexpected error %v, got %v", database.ErrNotFound, err)
 			}
-
-			currIT, err := store.GetCurrentStakerIterator() // check version 2
+			err = checkStakersContent(store, []Staker{}, current)
 			if err != nil {
-				return fmt.Sprintf("unexpected failure in staker iterator creation, error %v", err)
-			}
-			if currIT.Next() {
-				return fmt.Sprintf("expected empty iterator, got at least element %v", currIT.Value())
+				return err.Error()
 			}
-			currIT.Release()
 
 			// it's fine deleting unknown validator
 			store.DeleteCurrentValidator(&s)
-			_, err = store.GetCurrentValidator(s.SubnetID, s.NodeID) // check version 1
+			_, err = store.GetCurrentValidator(s.SubnetID, s.NodeID)
 			if err != database.ErrNotFound {
 				return fmt.Sprintf("unexpected error %v, got %v", database.ErrNotFound, err)
 			}
-
-			currIT, err = store.GetCurrentStakerIterator() // check version 2
+			err = checkStakersContent(store, []Staker{}, current)
 			if err != nil {
-				return fmt.Sprintf("unexpected failure in staker iterator creation, error %v", err)
+				return err.Error()
```

Comment:
could this simply be `return err`?

Reply:
I usually see the condition (i.e. this whole function) returning a string.
Here is what godoc has to say about it:
```
func prop.ForAll(condition interface{}, gens ...gopter.Gen) gopter.Prop
ForAll creates a property that requires the check condition to be true for all values, if the condition falsiies the generated values will be shrunk.

"condition" has to be a function with the same number of parameters as the provided generators "gens". The function may return a simple bool (true means that the condition has passed), a string (empty string means that condition has passed), a *PropResult, or one of former combined with an error.
```